src/Avalonia.Base/Data/StandardBindingBase.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.ComponentModel;
+using System.Globalization;
+using Avalonia.Controls;
+using Avalonia.Data.Converters;
+
+namespace Avalonia.Data;
+
+/// <summary>
+/// Defines properties common to both <see cref="Binding"/> and <c>CompiledBindingExtension</c>.
+/// </summary>
+public abstract class StandardBindingBase : BindingBase
```

Comment:
Not totally happy with this name. Another option I considered was `BindingExtensionBase` as it's the base for the two types of bindings commonly used as markup extensions, although it would not be totally accurate as 1. all classes can be used as markup extensions and 2. future hypothetical markup extensions may choose not to derive from this class.

Reply:
Imo, BindingExtensionBase is better anyway.

> 1. all classes can be used as markup extensions

This naming doesn't necessary imply that's one cannot use BindingBase directly in markup extensions. I think it fine.

> 2. future hypothetical markup extensions may choose not to derive from this class.

What would these _bindings_ extensions be? TypedBindingExtension? 
We can call it `UntypedBindingBase` if you will.

Reply:
Another option - just remove this class and copy paste its properties to concrete binding types. It's rather simple and essentially a choice of KISS over DRY.
That is unless you want to use `StandardBindingBase` in conditions elsewhere.

Reply:
Removed this class and copied the properties into the 2 final implementations. I didn't add an internal interface currently, not sure it's needed?